src/components/GenerateGifForm.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import './GenerateGifForm.css';
+
+class GenerateGifForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { name: '', titleText: '' };
+    this.handleInputUpdate = this.handleInputUpdate.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(evt) {
+    evt.preventDefault();
+    this.props.handleGenerateGIF();
+    this.props.updateGifFileName(this.state.name);
+    // set title
+    this.setState({ name: '', titleText: '' });
+  }
+
+  handleInputUpdate(evt) {
+    this.setState({
+      [evt.target.name]: evt.target.value
```

Comment:
here you'll need to dispatch an action rather than setting some state.

Reply:
Done